e2e-test-server/endtoendserver/scenarios.go

```diff
@@ -0,0 +1,112 @@
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/trace"
+	"google.golang.org/genproto/googleapis/rpc/code"
+)
+
+var scenarioHandlers = map[string]scenarioHandler{
+	"/basicPropagator": (*Server).unimplementedHandler,
+	"/basicTrace":      (*Server).basicTraceHandler,
+	"/complexTrace":    (*Server).complexTraceHandler,
+	"/health":          (*Server).healthHandler,
+}
+
+type scenarioHandler func(*Server, context.Context, request, *sdktrace.TracerProvider) *response
+
+type request struct {
+	scenario string
+	testID   string
+}
+
+type response struct {
+	statusCode code.Code
+	data       []byte
+	traceID    trace.TraceID
+}
+
+// healthHandler returns an OK response without creating any traces.
+func (s *Server) healthHandler(ctx context.Context, req request, tracerProvider *sdktrace.TracerProvider) *response {
+	return &response{statusCode: code.Code_OK}
+}
+
+// basicTraceHandler creates a basic trace and returns an OK response.
+func (s *Server) basicTraceHandler(ctx context.Context, req request, tracerProvider *sdktrace.TracerProvider) *response {
+	if req.testID == "" {
+		log.Printf("request is missing required field 'testID'. request: %+v", req)
+		return &response{
+			statusCode: code.Code_INVALID_ARGUMENT,
+			data:       []byte("request is missing required field 'testID'"),
+		}
+	}
+
+	tracer := tracerProvider.Tracer(instrumentingModuleName)
+	_, span := tracer.Start(ctx, "basicTrace",
+		trace.WithAttributes(attribute.String(testIDKey, req.testID)))
+	span.End()
+
+	return &response{statusCode: code.Code_OK, traceID: span.SpanContext().TraceID()}
+}
+
+// complexTraceHandler creates a complex trace and returns an OK response.
+func (s *Server) complexTraceHandler(ctx context.Context, req request, tracerProvider *sdktrace.TracerProvider) *response {
+	if req.testID == "" {
+		log.Printf("request is missing required field 'testID'. request: %+v", req)
+		return &response{
+			statusCode: code.Code_INVALID_ARGUMENT,
+			data:       []byte("request is missing required field 'testID'"),
+		}
+	}
+
+	tracer := tracerProvider.Tracer(instrumentingModuleName)
+	ctx, span := tracer.Start(ctx, "complexTrace/root",
+		trace.WithAttributes(attribute.String(testIDKey, req.testID)))
+
+	func(ctx context.Context) {
+		ctx, span := tracer.Start(ctx, "complexTrace/child1",
+			trace.WithSpanKind(trace.SpanKindServer),
+			trace.WithAttributes(attribute.String(testIDKey, req.testID)))
+		span.End()
+
+		func(ctx context.Context) {
+			_, span := tracer.Start(ctx, "complexTrace/child2",
+				trace.WithSpanKind(trace.SpanKindClient),
+				trace.WithAttributes(attribute.String(testIDKey, req.testID)))
+			span.End()
+		}(ctx)
+	}(ctx)
+
+	func(ctx context.Context) {
+		_, span := tracer.Start(ctx, "complexTrace/child3",
+			trace.WithAttributes(attribute.String(testIDKey, req.testID)))
+		span.End()
+	}(ctx)
+	span.End()
+
+	return &response{statusCode: code.Code_OK, traceID: span.SpanContext().TraceID()}
```

Comment:
to fix govet issue on golint
```suggestion
	ctx, rootSpan := tracer.Start(ctx, "complexTrace/root",
		trace.WithAttributes(attribute.String(testIDKey, req.testID)))
	defer rootSpan.End()

	func(ctx context.Context) {
		ctx, span := tracer.Start(ctx, "complexTrace/child1",
			trace.WithSpanKind(trace.SpanKindServer),
			trace.WithAttributes(attribute.String(testIDKey, req.testID)))
		span.End()

		func(ctx context.Context) {
			_, span := tracer.Start(ctx, "complexTrace/child2",
				trace.WithSpanKind(trace.SpanKindClient),
				trace.WithAttributes(attribute.String(testIDKey, req.testID)))
			span.End()
		}(ctx)
	}(ctx)

	func(ctx context.Context) {
		_, span := tracer.Start(ctx, "complexTrace/child3",
			trace.WithAttributes(attribute.String(testIDKey, req.testID)))
		span.End()
	}(ctx)

	return &response{statusCode: code.Code_OK, traceID: span.SpanContext().TraceID()}
```